File: CSharp/LeetCode.Test/007-ReverseInteger-Test.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace LeetCode.Test
{
    [TestClass]
    public class _007_ReverseInteger_Test
    {
        [TestMethod]
        public void GeneralTest_Positive()
        {
            var solution = new _007_ReverseInteger();
            var result = solution.Reverse(1);
            Assert.AreEqual(1, result);

            result = solution.Reverse(123);
            Assert.AreEqual(321, result);

            result = solution.Reverse(321);
            Assert.AreEqual(123, result);
        }

        [TestMethod]
        public void GeneralTest_Negative()
        {
            var solution = new _007_ReverseInteger();
            var result = solution.Reverse(-1);
            Assert.AreEqual(-1, result);

            result = solution.Reverse(-123);
            Assert.AreEqual(-321, result);

            result = solution.Reverse(-321);
            Assert.AreEqual(-123, result);
        }

        [TestMethod]
        public void GeneralTest_EqualZero()
        {
            var solution = new _007_ReverseInteger();
            var result = solution.Reverse(0);
            Assert.AreEqual(0, result);
        }

        [TestMethod]
        public void OverflowTest_Positive()
        {
            var solution = new _007_ReverseInteger();
            var result = solution.Reverse(1534236469);
            Assert.AreEqual(0, result);
        }

        [TestMethod]
        public void OverflowTest_Negative()
        {
            var solution = new _007_ReverseInteger();
            var result = solution.Reverse(-1534236469);
            Assert.AreEqual(0, result);
        }
    }
}

File: C++/LeetCode.Test/007-ReverseInteger-Test.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "..\LeetCode\\007-ReverseInteger.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace LeetCode;

namespace LeetCodeTest 
{
    TEST_CLASS(_007_ReverseInteger_Test)
    {
    public: 

        TEST_METHOD( ReverseTest_Positive)
        {
            _007_ReverseInteger solution;
            int result = solution.reverse(1);
            Assert::AreEqual(1, result);

            result = solution.reverse(123);
            Assert::AreEqual(321, result);

            result = solution.reverse(321);
            Assert::AreEqual(123, result);
        }

        TEST_METHOD(ReverseTest_Negative)
        {
            _007_ReverseInteger solution;
            int result = solution.reverse(-1);
            Assert::AreEqual(-1, result);

            result = solution.reverse(-123);
            Assert::AreEqual(-321, result);

            result = solution.reverse(-321);
            Assert::AreEqual(-123, result);
        }

        TEST_METHOD(ReverseTest_EqualZero)
        {
            _007_ReverseInteger solution;
            int result = solution.reverse(0);
            Assert::AreEqual(0, result);
        }

        TEST_METHOD(ReverseTest_PositiveOverflow)
        {
            _007_ReverseInteger solution;
            int result = solution.reverse(1534236469);
            Assert::AreEqual(0, result);
        }

        TEST_METHOD(ReverseTest_NegativeOverflow)
        {
            _007_ReverseInteger solution;
            int result = solution.reverse(-1534236469);
            Assert::AreEqual(0, result);
        }
    };
}